### ISSUE TITLE
Load relations not working

### ISSUE THREAD

**Issue by ilgala** [knuckleswtf/scribe#46]
- [x] **I confirm that I have read and attempted the tips in the [Troubleshooting Guide](https://scribe.readthedocs.io/en/latest/troubleshooting.html).**

**What happened?**
I'm trying to create the documentation for a listing endpoint where the returned collection models should also load some relations, here the doc block:

```php
/**
 * My model - index
 * bla bla, bla bla bla bla, bla.
 *
 * @group My Group
 * @queryParam XXX required description. Example: "XXX"
 * @queryParam YYY description. Example: "YYY"
 * @apiResourceCollection \My\Resource\Collection
 * @apiResourceModel \My\Resource\Model paginate=10 states=docs with=rel1,rel2,rel3
 *
 * @param \Illuminate\Http\Request $request
 *
 * @return \My\Resources\Collection
 */
public function index(Request $request) {
  // MY CODE
}
```

I expect something like:

```JSON
{
  "data": [
    [{
      "param": "xxx", 
      "param": "yyy", 
      "rel1": {
        "rel_param": "zzz"
      },
      "rel2": {
        "rel_param": "zzz"
      },
    }],
    // THE OTHER RESULTS
  ]
}
```
but in the generated documentation I only see:
```JSON
{
  "data": [
    [{
      "param": "xxx", 
      "param": "yyy"
    }],
    // THE OTHER RESULTS
  ]
}
```

**My environment:**
 - PHP version: 7.4.7
 - Laravel version: 6.18.20
 - Scribe Version: 1.1.0

**My Scribe config:**
```php
return [
    'type' => 'laravel',
    'static' => [
        'output_path' => 'public/docs',
    ],
    'laravel' => [
        'add_routes' => false,
        'docs_url' => '/docs',
        'middleware' => [],
    ],
    'auth' => [
        'enabled' => true,
        'in' => 'bearer',
        'name' => 'Authorization',
        'use_value' => env('SCRIBE_AUTH_KEY'),
        'extra_info' => 'You can retrieve your token by visiting your dashboard and clicking <b>Generate API token</b>.',
    ],
    'intro_text' => <<<'INTRO'
Welcome to DAPL API documentation!

<aside>As you scroll, you'll see code examples for working with the API in different programming languages in the dark area to the right (or as part of the content on mobile), and you can switch the programming language of the examples with the tabs in the top right (or from the nav menu at the top left on mobile).</aside>
INTRO
    ,
    'example_languages' => [
        'javascript',
        'php',
        'bash'
    ],
    'base_url' => env('APP_AUTH_URL'),
    'title' => null,
    'postman' => [
        'enabled' => true,
        'base_url' => null,
        'description' => null,
        'auth' => null,
    ],
    'default_group' => 'api',
    'logo' => resource_path('assets/logo.png'),
    'router' => 'laravel',
    'routes' => [
        [
            'match' => [
                'domains' => ['*'],
                'prefixes' => [
                    'api/*',
                ],
                'versions' => ['v1'],
            ],
            'include' => [],
            'exclude' => [],
            'apply' => [
                'headers' => [
                    'Content-Type' => 'application/json',
                    'Accept' => 'application/json',
                ],
                'response_calls' => [
                    'methods' => ['GET'],
                    'config' => [
                        'app.env' => 'testing',
                        'app.debug' => false,
                    ],
                    'cookies' => [],
                    'queryParams' => [],
                    'bodyParams' => [],
                    'fileParams' => [],
                ],
            ],
        ],
    ],
    'fractal' => [
        'serializer' => null,
    ],
    'faker_seed' => null,
    'strategies' => [
        'metadata' => [
            \Knuckles\Scribe\Extracting\Strategies\Metadata\GetFromDocBlocks::class,
        ],
        'urlParameters' => [
            \Knuckles\Scribe\Extracting\Strategies\UrlParameters\GetFromUrlParamTag::class,
        ],
        'queryParameters' => [
            \Knuckles\Scribe\Extracting\Strategies\QueryParameters\GetFromQueryParamTag::class
        ],
        'headers' => [
            \Knuckles\Scribe\Extracting\Strategies\Headers\GetFromRouteRules::class,
            \Knuckles\Scribe\Extracting\Strategies\Headers\GetFromHeaderTag::class,
        ],
        'bodyParameters' => [
            \Knuckles\Scribe\Extracting\Strategies\BodyParameters\GetFromFormRequest::class,
            \Knuckles\Scribe\Extracting\Strategies\BodyParameters\GetFromBodyParamTag::class,
        ],
        'responses' => [
            \Knuckles\Scribe\Extracting\Strategies\Responses\UseTransformerTags::class,
            \Knuckles\Scribe\Extracting\Strategies\Responses\UseResponseTag::class,
            \Knuckles\Scribe\Extracting\Strategies\Responses\UseResponseFileTag::class,
            \Knuckles\Scribe\Extracting\Strategies\Responses\UseApiResourceTags::class,
            \Knuckles\Scribe\Extracting\Strategies\Responses\ResponseCalls::class,
        ],
        'responseFields' => [
            \Knuckles\Scribe\Extracting\Strategies\ResponseFields\GetFromResponseFieldTag::class,
        ],
    ],
    'routeMatcher' => \Knuckles\Scribe\Matching\RouteMatcher::class,
];
```

**Comment by shalvah** [comment]
The question is [how are your models being generated](https://scribe.readthedocs.io/en/latest/documenting/documenting-endpoint-responses.html#how-model-instances-are-generated)? 

If Scribe is using your model factories, the related models won't be magically created unless your factory does it (or you use something like `$factory->afterCreating()` to create them). If Scribe is loading from your database, it doesn't create the related models either, it just adds the necessary queries to load them if they exist.

You can turn on debug logging with the `--verbose flag` to see which approach Scribe is using to generate your models.

**Comment by ilgala** [comment]
@shalvah I've made a factory state `docs` right for this purpose. An example:

```php
<?php

/** @var \Illuminate\Database\Eloquent\Factory $factory */

// Default factory
$factory->define(MyModel::class, function (Faker $faker) {
    return [
        'attribute' => $faker->attribute
        // [...]
    ];
});

// Almost identical to the one above
$factory->state(MyModel::class, 'docs', function (Faker $faker) {
    return [
        'attribute' => $faker->attribute
        // [...]
    ];
});

// [...]

$factory->afterCreatingState(MyModel::class, 'docs', function ($model, $faker) {
    $model->belongsTo->associate(new MyRelation([/* attributes */]));

   $model->hasMany->saveMany([
      [new MyRelation([/* attributes */])],
      [new MyRelation([/* attributes */])]
      // [...]
   ]);

    $model->push();
});
```

and as I wrote above the phpdocs has:

```
/**
 * My model - index
 * bla bla, bla bla bla bla, bla.
 *
 * [...]
 * 
 * @apiResourceModel \My\Resource\Model paginate=10 states=docs with=rel1,rel2,rel3
```

Should I add a `load` after `push`?

**Comment by shalvah** [comment]
I'm not sure what push() does. I think you should try my suggestion first, so you can be sure it's indeed your factory that's being used.

Also, wouldn't "belongsTo" and "has many" be replaced by the name of your relations?

**Comment by ilgala** [comment]
> I'm not sure what push() does. 

Have a look at the [official documentation](https://laravel.com/docs/6.x/eloquent-relationships#the-push-method) of the Laravel's version I'm using

> I think you should try my suggestion first, so you can be sure it's indeed your factory that's being used.

Ok, next time I have to add documentation I'll dig depper into your code and follow your suggestions. It may pass some times so please keep open the issue.

> Also, wouldn't "belongsTo" and "has many" be replaced by the name of your relations?

mine was an example... Pseudo code...

**Comment by shalvah** [comment]
Fixed in #104 (1.9.0)
